testing/integration/controller-system-buttons.nix

```diff
@@ -0,0 +1,131 @@
+pkgs.testers.runNixOSTest {
+  name = "Controller system calls";
+
+  nodes = {
+    playos = { config, ... }: {
+      imports = [
+        (import ../../base {
+          inherit pkgs kioskUrl playos-controller;
+          inherit safeProductName fullProductName greeting version;
+        })
+      ];
+
+      config = {
+        # These are sufficient to fool RAUC into thinking things are somewhat
+        # properly set up.
+        boot.kernelParams = [
+            "rauc.slot=a"
+        ];
+        boot.postBootCommands = ''
+            mkdir -p /boot/grub
+            ${pkgs.grub2_efi}/grub-editenv - set 'ORDER="a b"'
+            ${pkgs.grub2_efi}/grub-editenv - set a_TRY=0
+            ${pkgs.grub2_efi}/grub-editenv - set a_OK=1
+            ${pkgs.grub2_efi}/grub-editenv - set b_TRY=0
+            ${pkgs.grub2_efi}/grub-editenv - set b_OK=1
+        '';
+
+        fileSystems."/boot" = {
+          device = "tmpfs";
+          fsType = "tmpfs";
+          options = [ "mode=0755" ];
+        };
+
+
+        playos.selfUpdate = {
+          enable = true;
+          updateCert = pkgs.writeText "dummy.pem"  "";
+        };
+        services.connman = {
+          enable = pkgs.lib.mkOverride 0 true; # disabled in runNixOSTest by default
+        };
+        playos.storage = {
+          persistentDataPartition = {
+            device = "tmpfs";
+            fsType = "tmpfs";
+            options = [ "mode=0755" ];
+          };
+        };
+      };
+    };
+  };
+
+  # Note: using mostly `wait_for_console_text` mostly here, because test-driver
+  # crashes if using wait_for_unit or other higher level asserts
+  testScript = ''
+    def wait_for_http():
+        playos.wait_for_unit("playos-controller.service")
+        # wait_for_open_port fails due to some test-driver error
+        playos.wait_until_succeeds("curl --fail http://localhost:3333/")
+
+    # without these the test-driver crashes in later asserts for unclear reasons
+    def manual_restart():
+        playos.reboot()
+        # needed to avoid using wait_for_unit during reboot
+        playos.wait_for_console_text("NixOS Stage 2")
+        wait_for_http()
+
+
+    playos.start(allow_reboot=True)
+    wait_for_http()
+
+    # ===== Reboot works
+    with subtest("Reboot works"):
+        playos.succeed("curl -X POST http://localhost:3333/system/reboot >&2")
+        playos.wait_for_console_text("systemd.*The system will reboot now!")
+
+    manual_restart()
+
+    # ===== Factory reset works
+
+    # Ensure persistent-data-wipe service is running
+    with subtest("Factory reset works"):
+        playos.succeed("""
+            systemctl show --no-pager \
+                --property ActiveState \
+                playos-wipe-persistent-data.service | grep 'ActiveState=inactive'
+        """)
+        playos.succeed("curl -X POST http://localhost:3333/system/factory-reset >&2")
+        playos.wait_for_console_text("systemd.*Starting playos-wipe-persistent-data.service.")
```

Comment:
Maybe known, but this currently seems to fail, probably due to

`Error: # [   10.094713] playos-controller[726]: playos-controller: [ERROR] GUI Error: org.freedesktop.systemd1.NoSuchUnit: Unit boot.mount not found.`

Reply:
Fixed by c8082c988 and 71f2086de3